libraries/Microsoft.Bot.Builder.Core.Extensions/TranscriptLoggerMiddleware.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// When added, this middleware will log incoming and outgoing activitites to a ITranscriptStore 
+    /// </summary>
+    public class TranscriptLoggerMiddleware : IMiddleware
+    {
+        private static JsonSerializerSettings JsonSettings = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
```

Comment:
Consider using the same settings we use elsewhere for JSON Serialization (also used by the lowest-level connectors):

        public static readonly JsonSerializer BotMessageSerializer = JsonSerializer.Create(new JsonSerializerSettings
        {
            NullValueHandling = NullValueHandling.Ignore,
            Formatting = Formatting.Indented,
            DateFormatHandling = DateFormatHandling.IsoDateFormat,
            DateTimeZoneHandling = DateTimeZoneHandling.Utc,
            ReferenceLoopHandling = ReferenceLoopHandling.Serialize,
            ContractResolver = new ReadOnlyJsonContractResolver(),
            Converters = new List<JsonConverter> { new Iso8601TimeSpanConverter() }
        });